portfolio-project-matching-app/models/FilteredProject.js

```diff
@@ -1,10 +1,77 @@
 import { Project } from './Project'
 
 class FilteredProject {
+    // fields
+    project;
+    buttonFilters;
+    searchFiltered;
 
-    constructor(project, technology) {
-        this.project = project;
-        this.filters = [technology];
+    // static factory methods (used like overloaded constructors)
+    static fromButton(project, technology) {
```

Comment:
I've been liking these static factory methods, but really with JS allowed for constructor overloading like Dart does.